src/ngraph/op/topk.cpp

```diff
@@ -137,3 +137,182 @@ void op::TopK::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector&
 {
     throw ngraph_error("Forward-propagation-only operation");
 }
+
+const string op::v1::TopK::type_name{"TopK"};
+
+op::v1::TopK::TopK(const Output<Node>& data,
+                   const Output<Node>& k,
+                   const int64_t axis,
+                   const std::string& mode,
+                   const std::string& sort)
+    : Op{{data, k}}
+    , m_axis{axis}
+    , m_mode{mode_from_string(mode)}
+    , m_sort{sort_from_string(sort)}
+    , m_index_element_type{element::i32}
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v1::TopK::TopK(const Output<Node>& data,
+                   const Output<Node>& k,
+                   const int64_t axis,
+                   const Mode mode,
+                   const SortType sort)
+    : Op{{data, k}}
+    , m_axis{axis}
+    , m_mode{mode}
+    , m_sort{sort}
+    , m_index_element_type{element::i32}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::TopK::validate_and_infer_types()
+{
+    const auto& input_partial_shape = get_input_partial_shape(0);
+    const auto input_rank = input_partial_shape.rank();
+
+    NODE_VALIDATION_CHECK(this,
+                          input_rank.is_dynamic() || static_cast<size_t>(input_rank) > 0,
+                          "Input rank must be greater than 0.");
+
+    const auto& k_partial_shape = get_input_partial_shape(1);
+    NODE_VALIDATION_CHECK(this,
+                          k_partial_shape.is_static() && k_partial_shape.to_shape().size() == 0,
```

Comment:
I've used the `compatible()` method and it works but I had to use 0 instead of 1 because we're looking for a scalar here, not a vector with one element.

Reply:
Whoops! Yes, sorry, that's what I meant :)